tests/visualization_tests/test_pareto_front.py

```diff
@@ -31,35 +34,78 @@ def test_plot_pareto_front_2d(
     assert (figure.data[1]["x"] + figure.data[0]["x"]) == ()
     assert (figure.data[1]["y"] + figure.data[0]["y"]) == ()
 
-    # Test with three trials.
+    # Test with four trials.
+    study.enqueue_trial({"x": 1, "y": 2})
     study.enqueue_trial({"x": 1, "y": 1})
+    study.enqueue_trial({"x": 0, "y": 2})
     study.enqueue_trial({"x": 1, "y": 0})
-    study.enqueue_trial({"x": 0, "y": 1})
-    study.optimize(lambda t: [t.suggest_int("x", 0, 1), t.suggest_int("y", 0, 1)], n_trials=3)
+    study.optimize(lambda t: [t.suggest_int("x", 0, 2), t.suggest_int("y", 0, 2)], n_trials=4)
+
+    constraints_func: Optional[Callable[[FrozenTrial], Sequence[float]]]
+    if use_constraints_func:
+        # (x, y) = (1, 0) is infeasible; others are feasible.
+        def constraints_func(t: FrozenTrial) -> Sequence[float]:
+            if t.params["x"] == 1 and t.params["y"] == 0:
+                return [1.0]
+            else:
+                return [-1.0]
```

Comment:
Not a strong opinion, but can we shorten these lines as follows?
```suggestion
            return [1.0] if t.params["x"] == 1 and t.params["y"] == 0 else [-1.0]
```